pkg/metrics/metrics.go

```diff
@@ -225,21 +225,30 @@ var (
 		Help:      "Total number of successful policy regenerations",
 	})
 
+	// Deprecated: this metric will be removed in Cilium 1.6
 	// PolicyRegenerationTime is the total time taken to generate policies
 	PolicyRegenerationTime = prometheus.NewCounter(prometheus.CounterOpts{
 		Namespace: Namespace,
 		Name:      "policy_regeneration_seconds_total",
-		Help:      "Total sum of successful policy regeneration times",
+		Help:      "Total sum of successful policy regeneration times (Deprecated)",
 	})
 
+	// Deprecated: this metric will be removed in Cilium 1.6
 	// PolicyRegenerationTimeSquare is the sum of squares of total time taken
 	// to generate policies
 	PolicyRegenerationTimeSquare = prometheus.NewCounter(prometheus.CounterOpts{
 		Namespace: Namespace,
 		Name:      "policy_regeneration_square_seconds_total",
-		Help:      "Total sum of squares of successful policy regeneration times",
+		Help:      "Total sum of squares of successful policy regeneration times (Deprecated)",
 	})
 
+	// PolicyRegenerationTimeStats is the total time taken to generate policies
+	PolicyRegenerationTimeStats = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: Namespace,
+		Name:      "policy_regeneration_time_stats_seconds",
```

Comment:
nit: Is the "_stats" portion of this name meaningful/necessary?

Reply:
Yes, I also consider the "_stats" portion meaningless, but again, I wanted to be consistent with the endpoint regeneration metric which is called "endpoint_regeneration_time_stats_seconds". I'm fine with removing "_stats" portion from both of them.

Reply:
I see there's an argument there for consistency, but I agree that it's a bit of a pain to rename metrics. I guess it's better to be consistent than to minimize name length and this shouldn't cause any additional confusion around the metric meaning. I will withdraw my objection.